data-structures/binary_search_tree/bst.py

```diff
@@ -0,0 +1,109 @@
+class Node:
+    def __init__(self, value):
+        self.val = value
+        self.left = None
+        self.right = None
+
+
+class BST:
+    """
+    Binart Search Tree has 3 methods to sort nodes byself.
+    .inorder(), .preorder(), and .postorder()
+    """
+    def __init__(self, arr=()):
+        self.root = None
+        if len(arr):
+            for element in arr:
+                self.insert(element)
+
+    def in_order(self, operation):
+        """
+        In order traversal. Traverses nodes starting from head down to the left
+        node until val is None than continues to move down right and repeats.
+        """
+        def _walk(node=None):
+            """walk finction jump from node to node"""
+            if node is None:
+                return
+
+            if node.left is not None:
+                _walk(node.left)
+            operation(node.val)
+
+            if node.right is not None:
+                _walk(node.right)
+
+        if self.root is None:
+            return False
+        else:
+            _walk(self.root)
+
+    def pre_order(self, operation):
+        """
+        Preorder traversal. Traverses nodes from top to left then right.
+        Value of nodes is not considered.
+        """
+        def _walk(node=None):
+            if node is None:
+                return
+            operation(node.val)
+
+            if node.left is not None:
+                _walk(node.left)
+
+            if node.right is not None:
+                _walk(node.right)
+
+        if self.root is None:
+            return False
+        else:
+            _walk(self.root)
+
+    def post_order(self, operation):
+        """
+        Postorder traversal. Traverses nodes from bottom left to the right side
+        of the tree in order to collect all nodes in order then collects roots.
+        """
+        def _walk(node=None):
+            if node.left is not None:
+                _walk(node.left)
+
+            if node.right is not None:
+                _walk(node.right)
+            operation(node.val)
+
+            if node is None:
+                return
```

Comment:
This condition should be the first statement in your `_walk()` function since `node.left` will raise an exception if `node` is `None`.